app/user.go

```diff
@@ -2264,15 +2266,61 @@ func (a *App) ConvertBotToUser(bot *model.Bot, userPatch *model.UserPatch, sysad
 }
 
 func (a *App) GetThreadsForUser(userID, teamID string, options model.GetUserThreadsOpts) (*model.Threads, *model.AppError) {
-	threads, err := a.Srv().Store.Thread().GetThreadsForUser(userID, teamID, options)
-	if err != nil {
+	var result model.Threads
```

Comment:
I really like how this played out, with the app layer doing the parallelization, and keeping the sql store "simple". Something to aspire for more broadly?

Reply:
Agree.